### ISSUE TITLE
An invalid destination host was provided when using proxy-chains with connect

### ISSUE THREAD

**Issue by ofosukin** [JoshGlazebrook/socks#79]
I really do not know if this should be an issue or a cry for help with debugging 

**My Code**
`#!/usr/bin/env node

const SocksClient = require('socks').SocksClient;

const options = {
  destination: {
    host: 'ip-api.com', // host names are supported with SOCKS v4a and SOCKS v5.
    port: 80
  },
  command: 'connect', // Only the connect command is supported when chaining proxies.
  proxies: [ // The chain order is the order in the proxies array, meaning the last proxy will establish a connection to the destination.
    {
      host: '127.0.0.1', // ipv4, ipv6, or hostname
      port: 9050,
      type: 5
    },
    {
      host: 'x.x.x.x', // ipv4 address of my proxy concealed for security reasons
      port: 443,
      type: 5
    }
  ]
};

SocksClient.createConnectionChain(options)
.then(info => {
  console.log(info.socket);
  console.log(info.socket.remoteAddress) 
  info.socket.write('GET /json HTTP/1.1\nHost: ip-api.com\n\n');
  info.socket.on('data', (data) => {
    console.log(data.toString()); // ip-api.com sees that the last proxy in the chain (104.131.124.203) is connected to it.
  });
})
.catch(err => {
  // Handle errors
  console.log(err);
});`

**The Error**
{ Error: An invalid destination host was provided.
    at Object.validateSocksClientOptions (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/common/helpers.js:22:15)
    at Function.createConnection (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:39:19)
    at Function.<anonymous> (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:95:54)
    at Generator.next (<anonymous>)
    at /home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:4:12)
    at Promise (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:82:49)
    at new Promise (<anonymous>)
    at Function.createConnectionChain (/home/ofosukin-l/Documents/repositories/website_fingerprinting_theses_working/Ofosu/wfp_app-layer_defense/WFP_Splitting_Implementation/fingerprinting/splitting/SplitTor/node_modules/socks/build/client/socksclient.js:82:16)
  options:
   { command: 'connect',
     proxy: { host: '127.0.0.1', port: 9050, type: 4 },
     destination: { host: undefined, port: 443 } } }

As you can see, it seems the IP string assigned to the host parameter does not get parsed.
If anyone can help me ASAP, I will be grateful

**Comment by ofosukin** [comment]
After studying helpers.js, I could see that the isValidSocksProxy function checks the attribute host or ipaddress.
After changing this portion of my code above:

`proxies: [ // The chain order is the order in the proxies array, meaning the last proxy will establish a connection to the destination.
{
host: '127.0.0.1', // ipv4, ipv6, or hostname
port: 9050,
type: 5
},
{
host: 'x.x.x.x', // ipv4 address of my proxy concealed for security reasons
port: 443,
type: 5
}
]`

to this below:
`proxies: [ // The chain order is the order in the proxies array, meaning the last proxy will establish a connection to the destination.
{
ipaddress: '127.0.0.1', // ipv4, ipv6, or hostname
port: 9050,
type: 5
},
{
ipaddress: 'x.x.x.x', // ipv4 address of my proxy concealed for security reasons
port: 443,
type: 5
}
]`

The function validated my inputs and I could see a connection to my proxy.
**Basically, using attribute name 'ipaddress' instead of 'host' ** resolved the error.
However, the client seems to almost immediately close the connection with an F- Flag (tcpdump) sent to the proxy.


How can I force https requests over the chained proxies?

**Comment by JoshGlazebrook** [comment]
Okay part of this is actually a bug. The `ipaddress` field is carried over from older versions and technically both `host` or `ipaddress` should be enough, but it looks like it ignores the `host` field when connecting through chains.

**Comment by ofosukin** [comment]
I am new to nodejs as you can already tell. Could you please assist me on how I could use the created chained proxies with the request module to make an https request? or perhaps any directions
